api/pkg/gateway/gateway_test.go

```diff
@@ -0,0 +1,214 @@
+package gateway
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/labstack/echo/v4"
+	"github.com/shellhub-io/shellhub/pkg/models"
+)
+
+func TestTenantFromContext(t *testing.T) {
+	type Expected struct {
+		tenantID *models.Tenant
+	}
+
+	cases := []struct {
+		description string
+		headers     map[string]string
+		expected    Expected
+	}{
+		{
+			description: "verify if given value returns from header",
+			headers: map[string]string{
+				"X-Tenant-ID": "00000000-0000-4000-0000-000000000000",
+			},
+			expected: Expected{
+				tenantID: &models.Tenant{
+					ID: "00000000-0000-4000-0000-000000000000",
+				},
+			},
+		}, {
+			description: "validate empty tenant string, for ID behaviour",
+			headers: map[string]string{
+				"X-Tenant-ID": "",
+			},
+			expected: Expected{
+				tenantID: nil,
+			},
+		}, {
+			description: "validate empty tenant string, for ID behaviour",
+			headers: map[string]string{
+				"": "",
+			},
+			expected: Expected{
+				tenantID: nil,
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.description, func(t *testing.T) {
+			e := echo.New()
+			req := httptest.NewRequest(http.MethodPost, "/", nil)
+
+			req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+			for k, v := range tc.headers {
+				req.Header.Set(k, v)
+			}
+
+			rec := httptest.NewRecorder()
+			ctxEchoT := e.NewContext(req, rec)
+
+			gatewayCtxT := Context{
+				nil,
+				ctxEchoT,
+			}
+
+			ctx := context.WithValue(context.TODO(), "ctx", &gatewayCtxT) // nolint:revive
+
+			retTenantfctx := TenantFromContext(ctx)
+
+			require.Equal(t, tc.expected.tenantID, retTenantfctx)
+		})
+	}
+}
+
+func TestUsernameFromContext(t *testing.T) {
+	type Expected struct {
+		userNameTenant *models.Username
```

Comment:
Why `userNameTenant`?

Reply:
I'll be changed, it has been because of an old design decision